coralproject/talk
# [CORL-382] Allow moderators to ban a commenter from the comment stream

## What does this PR do?

- Adds a ban user drop down option in the comment stream's moderation drop down.
- Ban User has a loading state as it waits for a query response from the server when the drop down opens
- Fixes some optimistic update warnings
- `lookup` helper now supports resolving arrays with object references

## TODO
- [x] Refine loading state design
- [x] Reject comment when banning
- [x] Change copy to notify user that the comment will be rejected
- [x] Use default text to send email

## How do I test this PR?

- Login as a moderate capable user
- Open the comment stream
- Find a comment with a user you want to ban
- Click the moderation drop down via the top right caret on the comment
- Click the new Ban User option
- Confirm you want to ban them
- User will be banned and comment will be rejected
